vue/components/ui/atoms/image-item/image-item.vue

```diff
@@ -0,0 +1,278 @@
+        },
+        /**
+         * The name text alignment.
+         */
+        textAlign: {
+            type: String,
+            default: null
+        },
+        /**
+         * The props for the item button.
+         */
+        buttonProps: {
+            type: Object,
+            default: null
+        },
+        /**
+         * The action options to include in the
+         * dropdown.
+         */
+        optionsItems: {
+            type: Array,
+            default: () => []
+        },
+        /**
+         * If the item displays the
+         * highlight animation.
+         */
+        highlight: {
+            type: Boolean,
+            default: false
+        },
+        /**
+         * The duration of the highlight animation.
+         */
+        animationDuration: {
+            type: Number,
+            default: 3000
+        }
+    },
+    data: function() {
+        return {
+            highlightData: this.highlight,
+            optionsVisible: false
+        };
+    },
+    computed: {
+        optionsSlots() {
+            return Object.keys(this.$slots).filter(slot => slot.startsWith("options-"));
+        },
+        optionsScopedSlots() {
+            return Object.keys(this.$scopedSlots).filter(slot => slot.startsWith("options-"));
+        },
+        classes() {
+            const base = {};
+            base.highlight = this.highlightData;
+            return base;
+        },
+        style() {
+            const base = {};
+            if (this.highlightData) base["animation-duration"] = `${this.animationDuration}ms`;
+            if (this.height) base.height = `${this.height}px`;
+            if (this.width) base.width = `${this.width}px`;
+            return base;
+        },
+        imageStyle() {
+            const base = {};
+            if (this.height) base.height = `${this.height}px`;
+            if (this.width) base.width = `${this.width}px`;
+            return base;
+        },
+        nameStyle() {
+            const base = {};
+            if (this.width) base.width = `${this.width}px`;
+            if (this.textAlign) base["text-align"] = this.textAlign;
+            return base;
+        },
+        buttonIcon() {
+            return this.buttonProps.icon || "options";
+        }
+    },
+    watch: {
+        highlight(value) {
+            this.highlightData = value;
+        },
+        highlightData(value) {
+            this.$emit("update:highlight", value);
+        }
+    },
+    methods: {
+        onAnimationEnd() {
+            this.highlightData = false;
+        },
+        onClick(event) {
+            this.$emit("click", event);
+        },
+        onButtonClick(event) {
+            if (this.optionsItems.length === 0) {
+                this.$emit("click:button", event);
+                return;
+            }
+            this.optionsVisible = !this.optionsVisible;
+        },
+        onOptionsItemClick(item) {
+            this.$emit(`click:${item.event}`);

```

Comment:
```suggestion
            this.$emit(`click:option:${item.event}`);
```

Reply:
forwawrd the native event

Reply:
Can't forward the native event, the dropdown does no propagate it. Maybe that change is better in another PR, since it will change all clients of the dropdown.

Reply:
Nevermind then, forgot about that problem. We can't modify the dropdown event because it would be a breaking change, and adding the native event to the end of the argument list (non-breaking change) would be useless